comms/dht/src/crypt.rs

```diff
@@ -127,10 +125,25 @@ pub fn create_message_challenge_parts(
             buf
         })
         .unwrap_or_default();
-    mac_challenge.update(&e_pk);
 
-    mac_challenge.update(&body);
-    mac_challenge.finalize_fixed().into()
+    // we digest the given data into a domain independent hash function to produce a signature
+    // use of the hashing API for domain separation and deal with variable length input
+    let domain_separated_hash =
+        DomainSeparatedHasher::<Challenge, GenericHashDomain>::new(DOMAIN_SEPARATION_CHALLENGE_LABEL)
```

Comment:
```suggestion
        comms_dht_hash_domain().mac_hasher::<Challenge>()
```
Note that `mac_hasher` can also be used, which implements the `LengthExtensionAttackResistant` trait.